### PR TITLE
Automatically upload replays for locks in battles

### PR DESCRIPTION
**Pull request by panpawn**

### CODE REVIEW

File: chat-commands.js [hunk 1]
```diff
@@ -1760,6 +1760,9 @@ exports.commands = {
 		this.add(`|unlink|hide|${userid}`);
 		if (userid !== toId(this.inputUsername)) this.add(`|unlink|hide|${toId(this.inputUsername)}`);
 
+		// Automatically upload replays as evidence/reference to the punishment
+		if (room.type === 'battle') this.parse('/savereplay');
```
Comment (Zarel) [inline chat-commands.js]: Can you do this after global modlogging?

**Comment by panpawn** [comment]
Updated

File: chat-commands.js [hunk 2]
```diff
@@ -1762,6 +1762,9 @@ exports.commands = {
 
 		const globalReason = (target ? `: ${userReason} ${proof}` : ``);
 		this.globalModlog((week ? "WEEKLOCK" : "LOCK"), targetUser || userid, ` by ${user.name}${globalReason}`);
+
+		// Automatically upload replays as evidence/reference to the punishment
+		if (room.type === 'battle') this.parse('/savereplay');
```
Comment (Zarel) [inline chat-commands.js]: Also this'll need to be `if (room.battle)`

We don't currently support replays of non-battle games.

**Comment by panpawn** [comment]
Updated

**Comment by Zarel** [comment]
Looks good!

**Comment by Zarel** [comment]
Should this be a server setting? I'm guessing many servers are unregistered.

**Comment by Zarel** [comment]
On second thought I think it might be time to allow replay uploading on unregistered servers.

**Pull request merged**